src/lv_font/lv_font_loader.c

```diff
@@ -0,0 +1,680 @@
+        if(res != LV_FS_RES_OK) {
+            return -1;
+        }
+
+        gdsc->box_w = read_bits(&bit_it, header->wh_bits, &res);
+        if(res != LV_FS_RES_OK) {
+            return -1;
+        }
+
+        gdsc->box_h = read_bits(&bit_it, header->wh_bits, &res);
+        if(res != LV_FS_RES_OK) {
+            return -1;
+        }
+
+        int nbits = header->advance_width_bits + 2 * header->xy_bits + 2 * header->wh_bits;
+        int next_offset = (i < loca_count - 1) ? glyph_offset[i + 1] : (uint32_t)(glyph_length - 1);
+        int bmp_size = next_offset - glyph_offset[i] - nbits / 8;
+
+        if(i == 0) {
+            gdsc->adv_w = 0;
+            gdsc->box_w = 0;
+            gdsc->box_h = 0;
+            gdsc->ofs_x = 0;
+            gdsc->ofs_y = 0;
+        }
+
+        gdsc->bitmap_index = cur_bmp_size;
+        if(gdsc->box_w * gdsc->box_h != 0) {
+            cur_bmp_size += bmp_size;
+        }
+    }
+
+    uint8_t * glyph_bmp = (uint8_t *) lv_mem_alloc(sizeof(uint8_t) * cur_bmp_size);
+
+    font_dsc->glyph_bitmap = glyph_bmp;
+
+    cur_bmp_size = 0;
+
+    for(unsigned int i = 1; i < loca_count; ++i) {
+        lv_fs_res_t res = lv_fs_seek(fp, start + glyph_offset[i]);
+        if(res != LV_FS_RES_OK) {
+            return -1;
+        }
+        bit_iterator_t bit_it = init_bit_iterator(fp);
+
+        int nbits = header->advance_width_bits + 2 * header->xy_bits + 2 * header->wh_bits;
+
+        read_bits(&bit_it, nbits, &res);
+        if(res != LV_FS_RES_OK) {
+            return -1;
+        }
+
+        if(glyph_dsc[i].box_w * glyph_dsc[i].box_h == 0) {
+            continue;
+        }
+
+        int next_offset = (i < loca_count - 1) ? glyph_offset[i + 1] : (uint32_t)(glyph_length - 1);
+        int bmp_size = next_offset - glyph_offset[i] - nbits / 8;
+
+        for(int k = 0; k < bmp_size; ++k) {
+            glyph_bmp[cur_bmp_size + k] = read_bits(&bit_it, 8, &res);
+            if(res != LV_FS_RES_OK) {
+                return -1;
+            }
+        }
+        cur_bmp_size += bmp_size;
+    }
+    return glyph_length;
+}
+
+static bool lvgl_load_font(lv_fs_file_t * fp, lv_font_t * font)
+{
+    lv_font_fmt_txt_dsc_t * font_dsc = (lv_font_fmt_txt_dsc_t *)
+                                       lv_mem_alloc(sizeof(lv_font_fmt_txt_dsc_t));
+
+    memset(font_dsc, 0, sizeof(lv_font_fmt_txt_dsc_t));
+
+    font->dsc = font_dsc;
+
+    /* header */
+    int32_t header_length = read_label(fp, 0, "head");
+    if(header_length < 0) {
+        return false;
+    }
+
+    font_header_bin_t font_header;
+    if(lv_fs_read(fp, &font_header, sizeof(font_header_bin_t), NULL) != LV_FS_RES_OK) {
+        return false;
+    }
+
+    font->base_line = -font_header.descent;
+    font->line_height = font_header.ascent - font_header.descent;
+    font->get_glyph_dsc = lv_font_get_glyph_dsc_fmt_txt;
+    font->get_glyph_bitmap = lv_font_get_bitmap_fmt_txt;
+
+#if !(LVGL_VERSION_MAJOR == 6 && LVGL_VERSION_MINOR == 0)
```

Comment:
This code would never run as part of a 6.x system so I think this condition can be removed entirely.

Reply:
ok i'll change.

Reply:
done.